src/lang/cpp/stmt.rs

```diff
@@ -90,11 +90,14 @@ pub fn variable_declaration(node: &AST) -> DeclStmt {
                 variable_init_declaration(init_declarator, variable_type)
             }
             "array_declarator" => {
-                let rhs = expression(init_declarator).expect(&*format!(
-                    "Malformed array_declarator {:#?}",
-                    init_declarator
-                ));
-                DeclStmt::new(Some(variable_type), vec![rhs])
+                let rhs = match expression(init_declarator) {
+                    Some(expr) => vec![expr],
+                    None => {
+                        eprintln!("Malformed array_declarator {:#?}", init_declarator);
```

Comment:
I'm noticing a number of places where we just report that the AST was malformed and keep going. I take it that's going to be our default approach?

Reply:
Some of the nodes in the AST from treesitter can actually be malformed but you actually need to check whether that's the case or not. I think in reality most of the times the AST isn't malformed and it's just that the syntax can get pretty complex.

I'd say it's better to properly handle these cases in reality, I was just focusing on getting prophet and the parser to be able to communicate properly. I'll probably return to these different parts of the code to fix these at some point.